tools/integration_tests/multiwriters_and_multimounts/multiwriters_from_two_mounts_test.go

```diff
@@ -0,0 +1,185 @@
+	. "github.com/jacobsa/ogletest"
+)
+
+func getMountPath1() string {
+	return path.Join(setup.MntDir(), MountName1)
+}
+
+func getMountPath2() string {
+	return path.Join(setup.MntDir(), MountName2)
+}
+
+func getFilePathInMount1() string {
+	return path.Join(getMountPath1(), CommonFileName)
+}
+
+func getFilePathInMount2() string {
+	return path.Join(getMountPath2(), CommonFileName)
+}
+
+func getGCSObjPath() string {
+	return path.Join(setup.TestBucket(), CommonFileName)
+}
+
+func openFile(t *testing.T, filePath string) (file *os.File) {
+	file, err := os.OpenFile(filePath, os.O_RDWR|syscall.O_DIRECT, operations.FilePermission_0600)
+	if err != nil {
+		t.Errorf("error in opening file %s: %v", file.Name(), err)
+	}
+	return
+}
+
+func writeToFile(t *testing.T, file *os.File, fileContent string) {
+	_, err := file.WriteString(fileContent)
+	if err != nil {
+		t.Errorf("error in writing to the file %s: %v", file.Name(), err)
+	}
+}
+
+func syncFile(t *testing.T, file *os.File) {
+	err := file.Sync()
+	if err != nil {
+		t.Errorf("error in syncing file %s: %v", file.Name(), err)
+	}
+}
+
+func cleanAndCreateCommonFileInTestBucket(t *testing.T) {
+	// Cleaning one mount is sufficient as the test bucket is common.
+	err := operations.EmptyDir(getMountPath1())
+	if err != nil {
+		t.Errorf("Error while cleaning mount %v", err)
+	}
+
+	// Create file in mount 1.
+	_, err = os.Create(getFilePathInMount1())
+	if err != nil {
+		t.Errorf("Error creating common file %v", err)
+	}
+	// List files in mount 2 so that the common file starts reflecting in mount 2.
+	_, _ = os.ReadDir(getMountPath2())
+}
+
+func compareGCSObjContent(t *testing.T, objContent string) {
+	tempFile, err := os.CreateTemp(setup.MntDir(), "")
+	if err != nil {
+		t.Errorf("Error while creating temporary file %v", err)
+	}
+	tempFileName := tempFile.Name()
+	operations.CloseFile(tempFile)
+
+	err = operations.DownloadGcsObject(getGCSObjPath(), tempFileName)
+	if err != nil {
+		t.Errorf("Error downloading object from GCS %v", err)
+	}
+
+	setup.CompareFileContents(t, tempFileName, objContent)
+}
+
+func TestW1OpensW2OpensW1WritesFlushesW2WritesFlushes(t *testing.T) {
+	cleanAndCreateCommonFileInTestBucket(t)
+	//w1 opens
+	w1File := openFile(t, getFilePathInMount1())
+	defer operations.CloseFile(w1File)
+	//w2 opens
+	w2File := openFile(t, getFilePathInMount2())
+	defer operations.CloseFile(w2File)
+	dataToWrite1 := "sample text to write in file by W1\n"
+	// w1 writes
+	writeToFile(t, w1File, dataToWrite1)
+	// w1 flushes
+	syncFile(t, w1File)
+	// Check if content is flushed to GCS
+	compareGCSObjContent(t, dataToWrite1)
+
+	// w2 tries writing when w1 has already flushed. w2 should fail in the read
+	// that happens in the write flow to load file content from GCS into disk.
+	dataToWrite2 := "sample text to write in file by W2\n"
+	_, err := w2File.WriteString(dataToWrite2)
+
+	// Currently, the read flow returns EEXIST error when the object to be read is
```

Comment:
how is EEXIST is giving error message: no such file or directory.  can you link to the code

Reply:
Sorry I wanted to write ENOENT and not EEXIST.